packages/ember-views/lib/system/build-component-template.js

```diff
@@ -186,6 +211,25 @@ function normalizeComponentAttributes(component, isAngleBracket, attrs) {
     }
   }
 
+  // IE8 Support: IE8 cannot change the type attr of an input after it has been appended to
+  // any node. Therefore, we detect if the browser cannot change the type of the input after
+  // being appended, and unbind type.
+  if (normalized.type && !canChangeType) {
```

Comment:
The simple version does not address https://github.com/emberjs/ember.js/issues/12295. For the case where the component definition defines the type attribute instead of the template we need to compute the property value.